### PR TITLE
Fixup casing and add requested_env_vars to indexer config

### PR DESCRIPTION
**Pull request by eseliger**
Adds a missing property to the schema, and fixes it's casing, making sure that existing config will continue to work.

## Test plan

Added a test for unmarshalling.

### CODE REVIEW

File: enterprise/internal/codeintel/shared/types/index.go [hunk 1]
```diff
@@ -27,18 +27,59 @@ type Index struct {
 	Root               string                       `json:"root"`
 	Indexer            string                       `json:"indexer"`
 	IndexerArgs        []string                     `json:"indexer_args"` // TODO - convert this to `IndexCommand string`
+	RequestedEnvVars   []string                     `json:"requested_env_vars"`
 	Outfile            string                       `json:"outfile"`
 	ExecutionLogs      []executor.ExecutionLogEntry `json:"execution_logs"`
 	Rank               *int                         `json:"placeInQueue"`
 	AssociatedUploadID *int                         `json:"associatedUpload"`
```
Comment (Strum355) [inline enterprise/internal/codeintel/shared/types/index.go:34]: Hmmm casing :clueless:

Comment (eseliger) [reply enterprise/internal/codeintel/shared/types/index.go:34]: this is not exposed in the config, it's a weird mix of 2 uses of one type as far as I understood it

**Review by Strum355 — APPROVED** [review]

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.14 kb) | 0.00% (+0.14 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/es/fixup-indexer-config/compare-report.html;tab=live_object) for a full comparison between the commits b5ca6cd and 8cd16fe or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by Strum355** [comment]
Are we sure this isnt the one we want to change? :thinking: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@nsc/dependencies-list/-/blob/lib/codeintel/autoindex/config/types.go?L17

**Comment by eseliger** [comment]
no not sure at all :(

**Comment by efritz** [comment]
Fixed some of this at least.

**Comment by eseliger** [comment]
not relevant anymore.

**Pull request closed without merging**